### DISCUSSION TITLE
Custom date / time format in GenerateReleaseNotes extension?

### DISCUSSION THREAD

**Discussion by astanley** [rfennell/AzurePipelines#1896] [General]
Instead of the default date/time format, rendered by the GenerateReleaseNotes extension, I wish to display any time values using the following format: `YYYY-MM-DD HH:mm:ss`.  However I am unable to determine if custom formats is a capability of the extension, and if so how would I use them. 

Help?  Thanks in advance!

My template (snippet):
```
      ##Build - {{buildDetails.buildNumber}}
        - Date: {{buildDetails.startTime}}
```

Here's how my release notes look using the (default) date/time format for the extension:
<img width="593" alt="image" src="https://github.com/rfennell/AzurePipelines/assets/419930/bfe9ef62-73c4-43ba-a4fe-6bc388a4dafa">

Here's how I want my release notes to look using a custom format I define:
<img width="341" alt="image" src="https://github.com/rfennell/AzurePipelines/assets/419930/51bb3fd7-f43a-4b7e-b183-b40c21528081">

**Comment by rfennell** [comment]
This is a question that has been asked before, and the answer is you need to use a [custom Handlebars extension as detailed in the WIKI](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task#handlebar-extensions).

Below is a sample custom extension [taken from the sample folder](https://github.com/rfennell/AzurePipelines/blob/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203%20(and%20later)/), 

```
// A Handlebars Helper to allow bespoke date formatting
// Azure DevOps returns dates in the format "Mon Aug 09 2021 08:31:02 GMT+0000 (Coordinated Universal Time)"
// This functions swaps them to a local format e.g."August 09, 2021 - 08:31"
/* Usage
    {{date_formatter buildDetails.startTime}
*/

module.exports = {date_formatter(theDate) {
    return theDate.toLocaleString();
}};
```

You just need to 
1. Alter the actual JavaScript format line to get the date format you require
2. Put the custom extension as a file under source control or as an inline parameter for the task as per the [WIKI](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task#handlebar-extensions).
